growth/too/models.py

```diff
@@ -1007,3 +1012,247 @@ class Candidate(db.Model):
+        comment='Match with Gaia source')
+
+    gaia_parallax = db.Column(
+        db.Float,
+        nullable=True,
+        comment='Gaia parallax')
+
+    w4 = db.Column(
+        db.Float,
+        nullable=True,
+        comment='WISE W4')
+
+    w2mw3 = db.Column(
+        db.Float,
+        nullable=True,
+        comment='WISE W2-W3')
+
+    w1mw2 = db.Column(
+        db.Float,
+        nullable=True,
+        comment='WISE W1-W2')
+
+    jdstarthist = db.Column(
+        db.DateTime,
+        comment='Date (JD) of first detection')
+
+    ssmagnr = db.Column(
+        db.Float,
+        nullable=True,
+        comment='Nearest solar system object mag')
+
+    ssdistnr = db.Column(
+        db.Float,
+        nullable=True,
+        comment='Nearest solar system object distance')
+
+    CLU_sfr_ha = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU star formation rate Halpha')
+
+    CLU_mstar = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU stellar mass')
+
+    CLU_sfr_fuv = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU star formation rate fuv')
+
+    CLU_z = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU redshift')
+
+    CLU_name = db.Column(
+        db.String,
+        nullable=True,
+        comment='CLU name')
+
+    CLU_id = db.Column(
+        db.Integer,
+        nullable=True,
+        comment='CLU ID')
+
+    CLU_d_to_galaxy_arcsec = db.Column(
+        db.Float,
+        nullable=True,
+        comment='CLU distance arcsec')
+
+    comment = db.Column(
+        db.String,
+        nullable=True,
+        comment='Comment')
+
+    lightcurve = db.relationship(lambda: Lightcurve)
+
+
+class Lightcurve(db.Model):
+    """Candidate light curve pulled from the GROWTH
+    Marshal"""
+
+    name = db.Column(
+        db.ForeignKey(Candidate.name),
+        primary_key=True,
+        nullable=False,
+        comment='Candidate name')
+
+    date_observation = db.Column(
+        db.ARRAY(db.DateTime),
+        nullable=True,
+        comment='Observation date')
+
+    fil = db.Column(
+        db.ARRAY(db.String),
+        nullable=True,
+        comment='Filter')
+
+    instrument = db.Column(
+        db.ARRAY(db.String),
+        nullable=True,
+        comment='Instruments')
+
+    limmag = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Limiting magnitude')
+
+    mag = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Mag PSF')
+
+    magerr = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Mag uncertainty')
+
+    exptime = db.Column(
+        db.ARRAY(db.Float),
+        nullable=True,
+        comment='Exposure time')
+
+    programid = db.Column(
+        db.ARRAY(db.Integer),
+        nullable=True,
+        comment='Program ID number (1,2,3)')
+
+    first_detection_time_tmp = db.Column(
```

Comment:
What does "tmp" mean?

Reply:
Michael would rather avoid creating a column just for first detection time, but could not immediately find how to get that information directly from the query. So he preferred to leave "tmp" as temporary, to be removed in the future.

Reply:
_tmp columns removed in favor of sqlalchemy queries